packages/vats/tools/ibc-utils.js

```diff
@@ -0,0 +1,92 @@
+export const REMOTE_ADDR_RE =
+  /^(\/ibc-hop\/[^/]+)*\/ibc-port\/([^/]+)\/(ordered|unordered)\/([^/]+)$/s;
+harden(REMOTE_ADDR_RE);
+/** @typedef {`/${string}ibc-port/${string}/${'ordered' | 'unordered'}/${string}`} RemoteIbcAddress */
+
+export const LOCAL_ADDR_RE = /^\/ibc-port\/([-a-zA-Z0-9._+#[\]<>]+)$/;
+harden(LOCAL_ADDR_RE);
+/** @typedef {`/ibc-port/${string}`} LocalIbcAddress */
+
+/** @param {string} remoteAddr */
+export const decodeRemoteIbcAddress = remoteAddr => {
+  const match = remoteAddr.match(
+    /^(\/ibc-hop\/[^/]+)*\/ibc-port\/([^/]+)\/(ordered|unordered)\/([^/]+)$/s,
+  );
```

Comment:
Why not:
```suggestion
  const match = remoteAddr.match(REMOTE_ADDR_RE);
```
If that's not readable enough, we could switch to using [named capturing groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_backreferences#using_named_groups)

Reply:
sgtm. I was just moving this around but since we're touching it…

Reply:
good idea!